Update GDScript code style guide

NathanLovato opened the pull request:
This is a rewrite of our GDScript style guide, making all the guidelines up to date.

NathanLovato commented:
@EadhCaillin ready for review

EadhCaillin commented:
@NathanLovato
- 1- When you write "Ready for Review", are you referring to all the markdown files affected by the commit? Since some commits might be WIP, what do you think of specifiying "ALL" and if not, then specifying which files are ready for review?

- 2- What about previous files I didn't get to as part of the previous PR? Should I commit edits to this PR or a new one?

- 3- When you wrote the tutorial guidelines, you followed the structure described in the guidelines. In these code style guides, you don't exactly follow the structure. Is this intentional?

NathanLovato commented:
> 1- When you write "Ready for Review", are you referring to all the markdown files affected by the commit? 

Yes, I wouldn't ask you to review a work-in-progress. I generally write documents all at once. If you look at the changes, in two docs, I've just updated image file paths.

The only one to review is the biggest one.

> Should I commit edits to this PR or a new one?

It'd be best to use a new branch and separate PR. In part so we can merge edits as they're ready. If you could tackle this one in priority, that'd be ideal; the whole team and new teammates will need to review the changes.

> In these code style guides, you don't exactly follow the structure. Is this intentional?

Yes. This is like a technical reference document for our team. The tutorial structure template doesn't really fit. But if you have suggestions to improve it, they're welcome.

EadhCaillin commented:
I can see that only image paths were changed in 2 of the docs but having never seen the docs before, I couldn't tell if maybe they hadn't been reviewed in the past and I needed to do that.

Ok, for separate PRs. 
Ok for the priority.

NathanLovato commented:
All good. Those other two documents, I'm considering removing. Gotta find a moment to go over them again to see that.

We have plenty of old tuts and docs that could use a review and edits. But then you'd have an overwhelming amount of work on your plate! 🙂

More seriously, there's older content that's not where it could be, but it's not only writing - like in how valuable it is to people. Things that could use a makeover, others that aren't worth updating because they're useful to few persons.

EadhCaillin commented:
Accidentally pushed a commit to Master. Trying to undo it.

EadhCaillin commented:
Currently working on the right branch with the right up-to-date file. Sorry, I hadn't noticed I was on master. I don't know if you want to revert or undo and force push the commits, so I will leave it to you. I'll be more careful in the future. It's all still a bit new to me.

NathanLovato commented:
I just reverted the change. A quick note: whenever you get an error when trying to push, you should merge the branch. It's generally a sign there's something wrong happening - you're not on the right branch, or you don't have the latest version of the documents.

When that happens and you made a new commit already, you want to run `git pull --rebase`, which will download the latest changes and apply your new commit(s) on top of them.

To work on pull requests and with github, the github desktop app is pretty good if you're on windows. On other platforms, the [github cli](https://cli.github.com/) is really nice.

With it, to ensure you're on the right branch and editing the content of a pull request, you can run:

`gh pr checkout` followed by the pr number (last part of the page's URL) - for this one it'd be `gh pr checkout 265`

And you can list prs with `gh pr list`, which will list their number.

It's been a big time saver for me.

NathanLovato commented on content/docs/guidelines/best-practices/godot-gdscript/_index.md:
```diff
@@ -124,10 +124,10 @@ signal talk_finished
 
 {{% notice note %}}
-From Godot 3.2, you can write docstrings above any property, signal, or function as a series of comments above their definition, and the GDScript language server will show them in the docs completion. You can also use that to create a code reference with our tool [GDScript Docs Maker](https://github.com/GDQuest/gdscript-docs-maker).
+Starting from Godot 3.2, you can write docstrings as a series of comments above any class, property, signal, or function. The GDScript language server will show the comments as part of the description in the code completion box. You can also use that to create a code reference with our tool [GDScript Docs Maker](https://github.com/GDQuest/gdscript-docs-maker).
```
```suggestion
Starting from Godot 3.2, you can write docstrings as a series of comments above any class, property, signal, or function. The GDScript language server will show the comments as part of the description in the docs preview, if your code editor supports this feature. You can also use that to create a code reference with our tool [GDScript Docs Maker](https://github.com/GDQuest/gdscript-docs-maker).
```

NathanLovato replied:
This is specifically a docs preview feature that works over the language server, separate from auto-completion.

EadhCaillin commented:
I like this code style guide's flow and readability. This said, what do you think of creating a checklist that tutorial writers can refer to easily for each of the tutorial writing and code style guides? While the guides are useful as mandatory reading, I wonder if they are a quick enough check during work.

NathanLovato commented on content/docs/guidelines/best-practices/godot-gdscript/_index.md:
```diff
@@ -390,7 +390,7 @@ For other files, we just make sure to not leave spaces in the names. This is mos
 
 Always name your script and scene after the scene's root node if applicable. This makes it much easier to search for the files corresponding to a scene.
-
+<!-- "Name scripts and scenes after a scene's root node", should probably be included in the top list -->
```
```suggestion
```

NathanLovato commented on content/docs/guidelines/best-practices/godot-gdscript/_index.md:
```diff
@@ -334,109 +322,109 @@ signal started
+In short, the rules are:
 
-{{< figure src="img/naming-conventions-folders.png">}}
+- Name directories, scenes, and script files in `PascalCase`.
```
```suggestion
- Name directories, scenes, and script files in `PascalCase`.
- Name scripts and scenes after a scene's root node.
```

NathanLovato commented:
Yes definitely! Checklists are in my todo. Looking to do two or three. What do you think of this:

- One for code demos.
- One for outlines.
- One for complete writings?

Or would you do it differently?

Also, I just addressed your comment and tweaked a line. Is this good to merge?

Thanks for your work!

EadhCaillin commented:
Good to go

EadhCaillin commented:
@NathanLovato I feel the checklist should be one document maybe divided into collapsible sections. That's what I would want if I were writing. Now that I've read the tutorials, I just want a pre-flight checklist.
I would organize it in order of occurence in the project preparation or writing process (for example: "Before you start writing") . This way writers can progress down the list as they work. Within each section or level of progress, things could be further organized if necessary.
How interactive do we want it to be? Does it keep count or flag things that haven't been checked or is it a static tool?

NathanLovato commented:
I don't know how we could make it interactive with the way teammates work (part-time, with writing spread over days). I think a static checklist would do the job.
